server/accounts.go

```diff
@@ -2086,7 +2086,12 @@ func (a *Account) addServiceImportSub(si *serviceImport) error {
 	a.mu.Unlock()
 
 	cb := func(sub *subscription, c *client, acc *Account, subject, reply string, msg []byte) {
-		c.processServiceImport(si, acc, msg)
+		if didDeliver := c.processServiceImport(si, acc, msg); !didDeliver {
+			// We did not deliver to anything, so we will decrement the subs num messages.
+			sub.client.mu.Lock()
+			sub.nm--
```

Comment:
Hum... not sure if that will always works. For instance, we bumped the `sub.nm` only in the `if !traceOnly {` case in deliverMsg. Not sure if I can't not come up with a test case that would cause this to become negative...